src/components/checkbox-group/checkbox/bl-checkbox.ts

```diff
@@ -147,7 +154,9 @@ export default class BlCheckbox extends FormControlMixin(LitElement) {
           @blur=${this.blur}
         />
         <div class="check-mark">${icon ? html`<bl-icon name="${icon}"></bl-icon>` : null}</div>
-        <span class="label"><slot></slot></span>
+        <span class="label" ?hidden=${!this._hasLabel}
+          ><slot @slotchange=${this.handleSlotChange}></slot
+        ></span>
```

Comment:
I think moving `label` class to slot and removing `span` fixes the issue:

```html
<slot class="label"></slot>
```

No need for internal states and JS codes.